### PR TITLE
Added user data

### PR DESCRIPTION
**Pull request by goruha**
## What
* Added user data support

## Why 
* User data support required for some our projects

### CODE REVIEW

**Comment by comeanother** [comment]
Do we really need user data?
Why we don't use ansible for this purpose?

If the user data is changed, instance will be recreated.
But if we add changes to ansible playbook, instance will be re-provisioned without recreating.

**Comment by goruha** [comment]
@comeanother 
> Do we really need user data?
Yes we need.
I have project that does not use ansible. 
@osterman want to start us reuse common code in different projects.
Eventually I believe we can use ansible on all projects, but for now tf_instance should be flexible enough
to start using it.

> Why we don't use ansible for this purpose?
I do not  want to add additional tool for small functionality.
I do not want to redo existed user_data scripts to ansible to make provision consisted in project

> If the user data is changed, instance will be recreated.
> But if we add changes to ansible playbook, instance will be re-provisioned without recreating.
True. But even in current situation if you will change github auth keys params instance would be rebuilded.
Also I do not see any problem in such behaviour. 
I like that instances are immutable.

File: main.tf [hunk 1]
```diff
@@ -103,7 +113,7 @@ resource "aws_eip" "default" {
 
 # Apply the provisioner module for this resource
 module "ansible" {
-  source    = "git::https://github.com/cloudposse/tf_ansible.git?ref=tags/0.2.0"
+  source    = "git::https://github.com/cloudposse/tf_ansible.git"
```
Comment (drama17) [inline main.tf]: Why do you remove `?ref=tags/0.2.0`? If this tag do not corresponds to the latest commit in master let me know,  I can add new tag.

Comment (goruha) [inline main.tf]: Nice catch. Sec

Comment (goruha) [inline main.tf]: Done

**Pull request merged**